oneclick.py

```diff
@@ -658,17 +670,19 @@ def isbn_info_to_metadata(cls, book, include_bibliographic=True, include_formats
                 authors = book['authors']
                 for author in authors.split(";"):
                     sort_name = author.strip()
-                    roles = [Contributor.AUTHOR_ROLE]
-                    contributor = ContributorData(sort_name=sort_name, roles=roles)
-                    contributors.append(contributor)
+                    if sort_name:
+                        roles = [Contributor.AUTHOR_ROLE]
+                        contributor = ContributorData(sort_name=sort_name, display_name=sort_name, roles=roles)
+                        contributors.append(contributor)
 
             if 'narrators' in book:
                 narrators = book['narrators']
                 for narrator in narrators.split(";"):
                     sort_name = narrator.strip()
-                    roles = [Contributor.NARRATOR_ROLE]
-                    contributor = ContributorData(sort_name=sort_name, roles=roles)
-                    contributors.append(contributor)
+                    if sort_name:
+                        roles = [Contributor.NARRATOR_ROLE]
+                        contributor = ContributorData(sort_name=sort_name, display_name=sort_name, roles=roles)
```

Comment:
You shouldn't use a sort name as a display name. It's better to leave it blank and let the metadata wrangler fill it in. Is the absence of a display name causing problems?